packages/dashboard/src/components/tasks/task-inspector.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Grid,
+  Typography,
+} from '@mui/material';
+import { CardContent, useTheme } from '@mui/material';
+import { TaskEventLog, TaskState } from 'api-client';
+import React from 'react';
+import { AppControllerContext } from '../app-contexts';
+import { AppEvents } from '../app-events';
+import { RmfAppContext } from '../rmf-app';
+import { TaskInfo } from 'react-components';
+import { UserProfileContext } from 'rmf-auth';
+import { Enforcer } from '../permissions';
+import { TaskLogs } from './task-logs';
+
+export interface TableDataGridState {
+  task: TaskState | null;
+  open: boolean;
+  onClose: () => void;
+}
+
+export function TaskInspector({ task, open, onClose }: TableDataGridState): JSX.Element {
+  const theme = useTheme();
+  const rmf = React.useContext(RmfAppContext);
+  const appController = React.useContext(AppControllerContext);
+
+  const [taskState, setTaskState] = React.useState<TaskState | null>(null);
+  const [taskLogs, setTaskLogs] = React.useState<TaskEventLog | null>(null);
+
+  React.useEffect(() => {
+    if (!rmf || !task) {
+      setTaskState(null);
+      setTaskLogs(null);
+      return;
+    }
+    const sub = rmf.getTaskStateObs(task.booking.id).subscribe((subscribedTask) => {
+      (async () => {
+        try {
+          const logs = (
+            await rmf.tasksApi.getTaskLogTasksTaskIdLogGet(
+              subscribedTask.booking.id,
+              `0,${Number.MAX_SAFE_INTEGER}`,
+            )
+          ).data;
+          setTaskLogs(logs);
+        } catch {
+          console.log(`Failed to fetch task logs for ${subscribedTask.booking.id}`);
+          setTaskLogs(null);
+        }
+        setTaskState(subscribedTask);
+      })();
+    });
+    return () => sub.unsubscribe();
+  }, [rmf, task]);
+
+  React.useEffect(() => {
+    if (!rmf || !task) {
+      return;
+    }
+    const sub = rmf
+      .getTaskStateObs(task.booking.id)
+      .subscribe((subscribedTask) => setTaskState(subscribedTask));
+    return () => sub.unsubscribe();
+  }, [rmf, task]);
+
+  React.useEffect(() => {
+    AppEvents.taskSelect.next(task);
+  }, [task]);
```

Comment:
I don't think we need these lines anymore

Reply:
You're right!